core/src/main/java/com/amplifyframework/core/DeveloperMenuActivity.java

```diff
@@ -16,16 +16,112 @@
 package com.amplifyframework.core;
 
 import android.app.Activity;
+import android.content.Context;
+import android.hardware.Sensor;
+import android.hardware.SensorEvent;
+import android.hardware.SensorEventListener;
+import android.hardware.SensorManager;
 import android.os.Bundle;
+import android.view.KeyEvent;
+import android.view.View;
+
+import com.amplifyframework.util.Time;
 
 /**
  * This is the activity to display the developer menu.
  */
 public final class DeveloperMenuActivity extends Activity {
+    // A device movement is classified as a shake if the acceleration
+    // is above this threshold.
+    private static final double SHAKE_THRESHOLD = 11.7;
+    // The minimum duration (in milliseconds) that the device needs to
+    // be shaken in order to make the developer menu appear or disappear.
+    private static final int SHAKE_TIME = 500;
+
+    // Listen to accelerometer sensor events.
+    private final SensorEventListener sensorEventListener = new SensorEventListener() {
+        @Override
+        public void onSensorChanged(SensorEvent sensorEvent) {
+            float xAccel = sensorEvent.values[0]; // acceleration in the x-direction
+            float yAccel = sensorEvent.values[1]; // acceleration in the y-direction
+            float zAccel = sensorEvent.values[2]; // acceleration in the z-direction
+            double curAcceleration = Math.sqrt(((xAccel * xAccel) + (yAccel * yAccel) + (zAccel * zAccel)));
+            if (curAcceleration > SHAKE_THRESHOLD) {
+                long currentTime = Time.now();
+                if (shakeStart == 0) {
+                    shakeStart = currentTime;
+                } else if (currentTime - shakeStart > SHAKE_TIME) {
+                    changeVisibility();
+                    shakeStart = 0;
+                }
+            }
+        }
+
+        @Override
+        public void onAccuracyChanged(Sensor sensor, int accuracy) { }
+    };
+
+    // Manager for the device's sensors.
+    private SensorManager sensorManager;
```

Comment:
These should probably be listed above the event listener, especially since the listener references shakeStart.